cg/meta/workflow/analysis.py

```diff
@@ -511,13 +511,55 @@ def is_case_ready_for_analysis(self, case_id: str) -> bool:
             return False
         return True
 
+    def does_any_file_need_to_be_retrieved(self, case_id: str) -> bool:
+        """Checks whether we need to retrieve files from an external data location."""
+        if self._is_flow_cell_check_applicable(
+            case_id
+        ) and not self.status_db.are_all_flow_cells_on_disk(case_id):
+            LOG.warning(f"Case {case_id} is not ready - all flow cells not present on disk.")
+            return True
+        else:
+            if not self.are_all_spring_files_present(case_id):
+                LOG.warning(f"Case {case_id} is not ready - some files are archived.")
+                return True
+        return False
+
     def prepare_fastq_files(self, case_id: str, dry_run: bool) -> None:
-        """Retrieves or decompresses fastq files if needed, upon which an AnalysisNotReady error
+        """Retrieves or decompresses Spring files if needed. If so, an AnalysisNotReady error
         is raised."""
-        self.ensure_flow_cells_on_disk(case_id)
-        self.resolve_decompression(case_id, dry_run=dry_run)
+        self.ensure_files_are_present(case_id)
+        self.resolve_decompression(case_id=case_id, dry_run=dry_run)
         if not self.is_case_ready_for_analysis(case_id):
-            raise AnalysisNotReadyError("FASTQ file are not present for the analysis to start")
+            raise AnalysisNotReadyError("FASTQ files are not present for the analysis to start")
+
+    def ensure_files_are_present(self, case_id: str):
+        """Checks if any flow cells need to be retrieved and submits a job if that is the case.
+        Also checks if any spring files are archived and submits a job to retrieve any which are."""
+        self.ensure_flow_cells_on_disk(case_id)
+        if not self.are_all_spring_files_present(case_id):
+            LOG.warning(f"Files are archived for case {case_id}")
+            spring_archive_api = SpringArchiveAPI(
+                status_db=self.status_db,
+                housekeeper_api=self.housekeeper_api,
+                data_flow_config=self.config.data_flow,
+            )
+            spring_archive_api.retrieve_case(case_id)
+
+    def are_all_spring_files_present(self, case_id: str) -> bool:
+        """Return True if no Spring files are archived in the data location used by the customer tied to the given
+        case, and false otherwise."""
+        case: Case = self.status_db.get_case_by_internal_id(case_id)
+        for sample in [link.sample for link in case.links]:
+            if (
+                files := self.housekeeper_api.get_archived_files_for_bundle(
```

Comment:
```suggestion
                files :list[File] = self.housekeeper_api.get_archived_files_for_bundle(
```
Something got clipped here?

Reply:
Not clipped! It is the walrus operator which we can use ever since we upgraded to >= Python 3.8!